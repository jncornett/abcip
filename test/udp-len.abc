# the cks is hardcoded because otherwise it would be wrong
# since it would include all 40 bytes of payload.
d ( stack="eth:ip4:udp" )

# this has effectively 32 bytes of eth trailer
a ( ip4:tot=36; tcp:ulen=16, len=40, cks=0x2CA1 )

# similar to the above, but this time udp is correct 
# and ip4 has a shorter total length
a ( ip4:tot=36; tcp:len=40 )

