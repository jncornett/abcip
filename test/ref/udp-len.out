IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto UDP (17), length 36)
    10.1.2.3.48620 > 10.9.8.7.8: UDP, length 8
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0024 0001 0000 4011 5cb5 0a01 0203 0a09  .$....@.\.......
	0x0020:  0807 bdec 0008 0010 2ca1 4c6f 7265 6d20  ........,.Lorem.
	0x0030:  6970 7375 6d20 646f 6c6f 7220 7369 7420  ipsum.dolor.sit.
	0x0040:  616d 6574 2c20 636f 6e73 6563 7465 7475  amet,.consectetu
	0x0050:  7220                                     r.
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto UDP (17), length 36)
    10.1.2.3.48620 > 10.9.8.7.8: UDP, length 40
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0024 0002 0000 4011 5cb4 0a01 0203 0a09  .$....@.\.......
	0x0020:  0807 bdec 0008 0030 e406 6164 6970 6973  .......0..adipis
	0x0030:  6369 6e67 2065 6c69 742e 2045 7469 616d  cing.elit..Etiam
	0x0040:  2073 7573 6369 7069 742c 2071 7561 6d20  .suscipit,.quam.
	0x0050:  6174                                     at
